app/src/main/java/com/example/evently/ui/common/LiveEventsFragment.java

```diff
@@ -0,0 +1,73 @@
+package com.example.evently.ui.common;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import android.content.Context;
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.Fragment;
+import androidx.lifecycle.LiveData;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.example.evently.R;
+import com.example.evently.data.model.Event;
+
+/**
+ * Fragment base class that renders a live-updating list of {@link Event} objects.
+ * <p>
+ * Subclasses supply a {@link LiveData} source and refresh trigger, and provide
+ * per-item click handling via {@link #onEventClick(Event)}.
+ */
+public abstract class LiveEventsFragment extends Fragment {
```

Comment:
Take a look at the formulation I posted in discord. It's actually more general than this and can incorporate a lot more reusability.